cupy/fft/fft.py

```diff
@@ -357,7 +357,7 @@ def _exec_fftn(a, direction, value_type, norm, axes, overwrite_x,
         raise NotImplementedError('Only C2C and Z2Z are supported.')
 
     if a.base is not None:
```

Comment:
I confirmed this patch can fix both #3033 and #3079 and also pass all tests:
```suggestion
    if a.base is not None \
       and not a.flags.c_contiguous and not a.flags.f_contiguous:
```
Should we do this @grlee77?

Reply:
Yeah, I think so. By the way, there is also `a.flags.forc` which should be True if it is either C or Fortan contiguous.

Reply:
```Python
if a.base is not None and not a.flags.forc:
```

Reply:
Oh that's better! I didn't notice this flag...

Reply:
This change has now been made along with a test case adapted from your example in #3079